README.md

```diff
@@ -138,6 +143,74 @@ and then propogate or discard the resulting
 `TypeError: descriptor 'isdigit' for 'str' objects doesn't apply to a 'int' object`
 exception.  We encourage libraries to document the behaviour they choose.
 
+### Consuming and creating metadata
+
+All `annotated-types` metadata should inherit from one of `BaseMetadata` or `GroupedMetadata`, which are base classes provided by `annotated-types`.
+
+In this section we discuss how each one of these classes can be used and how to use them when consuming metadata.
+
+#### `BaseMetadata`
+
+We provide `BaseMetadata` as base class for all metadata that serves as a namespace and identifier to easily filter metadata from this package out from other metadata that may be contained in the `Annotated[...]` type annotation.
+
+If you are creating your own metadata, you _should_ inherit from `annotated_types.BaseMetadata`:
+
+```python
+from dataclasses import dataclass
+from annotated_types import BaseMetadata, GroupedMetadata
+
+@dataclass
+class Description(BaseMetadata):
+    """Description of a data type/field"""
+    description: str
+
+@dataclass
+class Pattern(BaseMetadata):
+    """Regex pattern constraint"""
+    pattern: str
+```
+
+Note that you do not have to inherit from `BaseMetadata` if you are creating a grouping of `BaseMetadata`.
+In that case you should inherit from `GroupedMetadata` so that the grouping itself is not considered metadata.
```

Comment:
```suggestion
In that case you should inherit from `GroupedMetadata` so that the grouping itself is not considered metadata (it is a _collection_ of metadata).
```